sync-core/src/main/java/com/cloudant/mazha/CouchClient.java

```diff
@@ -120,6 +125,34 @@ private InputStream executeToInputStream(HttpConnection connection) throws Couch
         }
     }
 
+    private <T> T executeWithRetry(final Callable<T> task) {
+        RetriableTask<T> retriableTask = new RetriableTask<T>(10, 10, new Callable<T>() {
+            @Override
+            public T call() throws Exception {
+                logger.fine("executeWithRetry: call()");
+                return task.call();
+            }
+        });
+        try {
+            return retriableTask.call();
+        } catch (RetryException re) {
+            // unpack cause from RetryException
+            // if it's a CouchException, throw as-is
+            // anything else, return null
+            try {
+                throw re.getCause();
+            } catch (CouchException ce) {
+                logger.fine("Re-throwing CouchException from executeWithRetry: "+ce);
+                throw ce;
+            } catch (Throwable e) {
+                return null;
```

Comment:
Do we need to log these non-`CouchException` cases? Even if they are unlikely it might be worth having a warning in the log or something so we can figure out why a null might have come back.